src/Raven.Studio/typescript/viewmodels/database/settings/databaseSettings.ts

```diff
@@ -0,0 +1,526 @@
+        let customizedEntries = 0;
+        
+        this.allEntries().forEach(x => {
+            if (x instanceof models.databaseEntry && x.data.Metadata.Category === category && x.override()) {
+                customizedEntries++;
+            }
+        });
+        
+        return customizedEntries;
+    }
+    
+    cancelEdit() {
+        if (this.dirtyFlag().isDirty()) {
+            this.confirmUnsavedchanges()
+                .then(result => {
+                    if (result.can) {
+                        this.exitEditMode();
+                    }
+                });
+        } else {
+            this.exitEditMode();
+        }
+    }
+
+    confirmUnsavedchanges() {
+        const text = `<div class="padding">
+                         <div class='bg-warning text-warning padding padding-xs margin-right'><i class='icon-warning margin-right margin-right-sm'></i>You have unsaved changes !</div>
+                         <div class="margin-top">Clicking <strong>OK</strong> will reload all database settings from the server <strong>without saving</strong> your changes.</div>
+                      </div>`;
+
+        return this.confirmationMessage("Are you sure?", text,
+            {
+                html: true,
+                buttons: ["Cancel", "OK"]
+            });
+    }
+    
+    exitEditMode() {
+        this.viewMode("summaryMode");
+        this.fetchData();
+        this.computeEntriesToShow();
+    }
+    
+    switchToEditMode() {
+        if (this.editModeHasBeenEntered) {
+            this.viewMode("editMode");
+            this.setupEditMode();
+        } else {
+            this.confirmationMessage("Are you an expert?",
+                "Modify the database settings only if you know what you are doing!", { buttons: ["Cancel", "OK"] })
+                .done(() => {
+                    this.editModeHasBeenEntered = true;
+                    this.viewMode("editMode");
+                    this.initCategoryGrid();
+                    this.setupEditMode(false);
+                });
+        }
+    }
+    
+    private setupEditMode(reset: boolean = true) {
+        this.dirtyFlag().reset();
+        
+        if (reset) {
+            this.categoriesGridController().reset(false);
+        }
+        
+        this.setCategory(this.selectedCategory() || this.allCategoryNames()[0]);
+        this.computeEntriesToShow();
+       
+        this.allEntries().forEach(entry => {
+            entry.showEntry(this.shouldShowEntry(entry));
+
+            if (entry instanceof models.databaseEntry) {
+                entry.override.subscribe((override) => {
+                    
+                    const category = this.categoriesInfo().find(x => x.name() === entry.data.Metadata.Category);
+                    const currentCustomizedCount = category.customizedEntriesCount();
+                    if (override) {
+                        category.customizedEntriesCount(currentCustomizedCount+1)
```

Comment:
I wonder if customizedEntriesCount should be observable to avoid need for manually tracking that?

Reply:
+ missing ; at the end fo lines

Reply:
done - changed to observable